src/slug.rs

```diff
@@ -0,0 +1,77 @@
+//! A trait (and implementation) to specify the anchor calculation.
+use std::{borrow::Cow, collections::HashMap};
+
+use once_cell::sync::Lazy;
+use regex::Regex;
+
+pub trait Slugify {
+    fn slugify(&mut self, str: &str) -> Cow<str>;
```

Comment:
Can you make it like this to not tie the lifetime of the slugifier to the cow

```
    fn slugify<'a>(&mut self, str: &'a str) -> Cow<'a, str>;
```

Reply:
Ah, good catch! Done.